cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -18,48 +18,50 @@
 
 import co.cask.cdap.api.schedule.SchedulableProgramType;
 import co.cask.cdap.api.schedule.Schedule;
+import co.cask.cdap.api.schedule.ScheduleSpecification;
+import co.cask.cdap.app.ApplicationSpecification;
 import co.cask.cdap.app.runtime.ProgramRuntimeService;
 import co.cask.cdap.app.store.Store;
 import co.cask.cdap.app.store.StoreFactory;
 import co.cask.cdap.config.PreferencesStore;
+import co.cask.cdap.internal.schedule.StreamSizeSchedule;
 import co.cask.cdap.internal.schedule.TimeSchedule;
 import co.cask.cdap.proto.Id;
-import com.google.common.base.Preconditions;
 import com.google.common.base.Supplier;
 import com.google.common.base.Throwables;
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
 import com.google.common.util.concurrent.AbstractIdleService;
-import org.quartz.CronScheduleBuilder;
-import org.quartz.Job;
-import org.quartz.JobBuilder;
-import org.quartz.JobDetail;
-import org.quartz.JobKey;
-import org.quartz.Scheduler;
 import org.quartz.SchedulerException;
-import org.quartz.Trigger;
-import org.quartz.TriggerBuilder;
-import org.quartz.TriggerKey;
-import org.quartz.spi.JobFactory;
-import org.quartz.spi.TriggerFiredBundle;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 /**
- * Abstract scheduler service common scheduling functionality. The extending classes should implement
- * prestart and poststop hooks to perform any action before starting the quartz scheduler and after stopping
- * the quartz scheduler.
+ * Abstract scheduler service common scheduling functionality. For each {@link Schedule} implementation, there is
+ * a scheduler that this class will delegate the work to.
+ * The extending classes should implement prestart and poststop hooks to perform any action before starting all
+ * underlying schedulers and after stopping them.
  */
 public abstract class AbstractSchedulerService extends AbstractIdleService implements SchedulerService {
 
   private static final Logger LOG = LoggerFactory.getLogger(AbstractSchedulerService.class);
-  private final WrappedScheduler delegate;
-
-  public AbstractSchedulerService(Supplier<org.quartz.Scheduler> schedulerSupplier, StoreFactory storeFactory,
-                                  ProgramRuntimeService programRuntimeService, PreferencesStore preferencesStore) {
-    this.delegate = new WrappedScheduler(schedulerSupplier, storeFactory, programRuntimeService, preferencesStore);
+  private final TimeScheduler timeScheduler;
+  private final StreamSizeScheduler streamSizeScheduler;
+  private final StoreFactory storeFactory;
+
+  private Store store;
+
+  public AbstractSchedulerService(Supplier<org.quartz.Scheduler> schedulerSupplier,
```

Comment:
From the javadoc of the scheduler supplier: "Provides a supplier of quartz scheduler so that initialization of the scheduler can be done after guice injection." 
We don't require that the initialization be made after guice injection for the ``StreamSizeScheduler``. The call to ``start`` will actually make the initialization.